pkg/frame/inplace/read_test.go

```diff
@@ -0,0 +1,219 @@
+									Data:          [][]int{{1}, {2}, {3}, {0}},
+								},
+							},
+						},
+					}),
+				},
+			},
+		},
+		{
+			Name: "2x2, 2 frames, 2 samples/pixel",
+			existingData: dicom.Dataset{
+				Elements: []*dicom.Element{
+					mustNewElement(t, tag.MediaStorageSOPClassUID, []string{"1.2.840.10008.5.1.4.1.1.1.2"}),
+					mustNewElement(t, tag.MediaStorageSOPInstanceUID, []string{"1.2.3.4.5.6.7"}),
+					mustNewElement(t, tag.TransferSyntaxUID, []string{uid.ImplicitVRLittleEndian}),
+					mustNewElement(t, tag.Rows, []int{2}),
+					mustNewElement(t, tag.Columns, []int{2}),
+					mustNewElement(t, tag.NumberOfFrames, []string{"2"}),
+					mustNewElement(t, tag.BitsAllocated, []int{16}),
+					mustNewElement(t, tag.SamplesPerPixel, []int{2}),
+
+					mustNewElement(t, tag.PixelData, dicom.PixelDataInfo{
+						IsEncapsulated: false,
+						Frames: []*frame.Frame{
+							{
+								Encapsulated: false,
+								NativeData: frame.NativeFrame{
+									BitsPerSample: 16,
+									Rows:          2,
+									Cols:          2,
+									Data:          [][]int{{1, 2}, {3, 2}, {1, 2}, {3, 2}},
+								},
+							},
+							{
+								Encapsulated: false,
+								NativeData: frame.NativeFrame{
+									BitsPerSample: 16,
+									Rows:          2,
+									Cols:          2,
+									Data:          [][]int{{1, 2}, {3, 2}, {1, 2}, {3, 5}},
+								},
+							},
+						},
+					}),
+				},
+			},
+		},
+		{
+			Name: "1x1, 3 frames, 3 samples/pixel, data bytes with padded 0",
+			existingData: dicom.Dataset{
+				Elements: []*dicom.Element{
+					mustNewElement(t, tag.MediaStorageSOPClassUID, []string{"1.2.840.10008.5.1.4.1.1.1.2"}),
+					mustNewElement(t, tag.MediaStorageSOPInstanceUID, []string{"1.2.3.4.5.6.7"}),
+					mustNewElement(t, tag.TransferSyntaxUID, []string{uid.ImplicitVRLittleEndian}),
+					mustNewElement(t, tag.Rows, []int{1}),
+					mustNewElement(t, tag.Columns, []int{1}),
+					mustNewElement(t, tag.NumberOfFrames, []string{"3"}),
+					mustNewElement(t, tag.BitsAllocated, []int{8}),
+					mustNewElement(t, tag.SamplesPerPixel, []int{3}),
+
+					mustNewElement(t, tag.PixelData, dicom.PixelDataInfo{
+						IsEncapsulated: false,
+						Frames: []*frame.Frame{
+							{
+								Encapsulated: false,
+								NativeData: frame.NativeFrame{
+									BitsPerSample: 8, Rows: 1, Cols: 1,
+									Data: [][]int{{1, 2, 3}},
+								},
+							}, {
+								Encapsulated: false,
+								NativeData: frame.NativeFrame{
+									BitsPerSample: 8, Rows: 1, Cols: 1,
+									Data: [][]int{{4, 5, 6}},
+								},
+							}, {
+								Encapsulated: false,
+								NativeData: frame.NativeFrame{
+									BitsPerSample: 8, Rows: 1, Cols: 1,
+									Data: [][]int{{7, 8, 9}},
+								},
+							},
+						},
+					}),
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.Name, func(t *testing.T) {
+			var filesOut bytes.Buffer
+			require.NoError(t, dicom.Write(io.Writer(&filesOut), tc.existingData))
```

Comment:
nit: let's not use require/assert, and instead just use Go code to do comparisons, see https://google.github.io/styleguide/go/decisions#assert

Reply:
Removed at 16701faf01880474451f742d3a8c7b4aff370b61